### PR TITLE
Introduce larger buckets for request_filter_duration_seconds and request_wait_duration_seconds

### PR DESCRIPTION
**Pull request by andrewsykim**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

kube-apiserver currently uses max bucket of 5s for request_filter_duration_seconds and 30s for request_wait_duration_seconds. Comparing filter latency for type `priorityandfairness` with `request_wait_duration_seconds` can be misleading because queue wait times are configured to exceed 5s by default. In addition, the default request wait limit is 15s (default request timeout of 60 / 4), but since there is only a 10s and 30s bucket, requests reaching the default wait limit of 15s are being reported as waiting 30s.

This PR improves usability of these metrics by:
* adding a 10s, 15s and 30s bucket to request_filter_duration_seconds
* adding a 15s bucket to request_wait_duration_seconds

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
- Update apiserver metric request_filter_duration_seconds to include a 10s, 15s and 30s bucket.
- Update apiserver metric request_wait_duration_seconds to include a 15s bucket.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ var (
 			Subsystem:      APIServerComponent,
 			Name:           "request_filter_duration_seconds",
 			Help:           "Request filter latency distribution in seconds, for each filter type",
-			Buckets:        []float64{0.0001, 0.0003, 0.001, 0.003, 0.01, 0.03, 0.1, 0.3, 1.0, 5.0},
+			Buckets:        []float64{0.0001, 0.0003, 0.001, 0.003, 0.01, 0.03, 0.1, 0.3, 1.0, 5.0, 10.0, 15.0, 30.0},
```
Comment (andrewsykim) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:232]: (I haven't updated metrics in a while, I'm assuming we're not breaking compatibility if we only update the buckets)

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ const (
 
 var (
 	queueLengthBuckets            = []float64{0, 10, 25, 50, 100, 250, 500, 1000}
-	requestDurationSecondsBuckets = []float64{0, 0.005, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30}
+	requestDurationSecondsBuckets = []float64{0, 0.005, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 15, 30}
```
Comment (andrewsykim) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go:50]: 15s is based on the default request wait limit (default request timeout of 60 / 4): https://github.com/kubernetes/kubernetes/blob/master/pkg/controlplane/apiserver/config.go#L193

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go (hunk 1 above)
Comment (andrewsykim) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:232]: Reason for 10, 15 and 30 is just to match largest buckets in requestDurationSecondsBuckets

**Comment by wojtek-t** [comment]
LGTM

but I want someone from SIG instrumentatil to lgtm too

/assign @dgrisonnet

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve

**Pull request merged**